analyses/collapse-rnaseq/run-collapse-rnaseq.sh

```diff
@@ -19,17 +19,58 @@ cd "$script_directory" || exit
 # create results directory if it doesn't already exist
 mkdir -p results
 
+# Usage: project acronym to use as prefix for input out files 
+usage(){ echo "Usage: $0 [-h] [-p <project acronym>] [-l <library strategy>] [-q <quantification type>] " 1>&2; exit 1; }
+
+while getopts ":hl:p:q:" opt; do
+    case "${opt}" in
+	h)
+	    usage
+	    ;;
+	p)
+	    project_acronym=$OPTARG
+	    ;;
+	l)
+	    libraryStrategies=$OPTARG
+	    ;;
+        q)
+            quantificationType=$OPTARG
+            ;;
+	:)
+	    printf "missing argument for -%s\n" "$OPTARG" 1>&2
+	    usage
+	    ;;
+	\?)
+	    printf "illegal option: -%s\n" "$OPTARG" 1>&2
+	    usage
+	    ;;
+    esac
+done
+shift $((OPTIND - 1))
+
+if [ -z "${project_acronym}" ]; then
+    usage
+fi
+
+if [ -z "${libraryStrategies}" ]; then
+    usage
+fi
+
+if [ -z "${quantificationType}" ]; then
+    usage
+fi
+
 # generate collapsed matrices for poly-A and stranded datasets
-libraryStrategies=("polya" "stranded")
+
 for strategy in ${libraryStrategies[@]}; do
 
   Rscript --vanilla 01-summarize_matrices.R \
-    -i ../../data/pbta-gene-expression-rsem-fpkm.${strategy}.rds \
+    -i ../../data/${project_acronym}-gene-expression-rsem-${quantificationType}.${strategy}.rds \
     -g ../../data/gencode.v27.primary_assembly.annotation.gtf.gz \
-    -m results/pbta-gene-expression-rsem-fpkm-collapsed.${strategy}.rds \
-    -t results/pbta-gene-expression-rsem-fpkm-collapsed_table.${strategy}.rds
+    -m results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed.${strategy}.rds \
+    -t results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed_table.${strategy}.rds\
 
 done
 
 # run the notebook for analysis of dropped genes
-Rscript -e "rmarkdown::render(input = '02-analyze-drops.Rmd', params = list(polya.annot.table = 'results/pbta-gene-expression-rsem-fpkm-collapsed_table.polya.rds', stranded.annot.table = 'results/pbta-gene-expression-rsem-fpkm-collapsed_table.stranded.rds'), clean = TRUE)"
+Rscript -e "rmarkdown::render(input = '02-analyze-drops.Rmd', params = list(polya.annot.table = 'results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed_table.polya.rds', stranded.annot.table = 'results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed_table.stranded.rds'), clean = TRUE)"
```

Comment:
Thanks for the updated. However, we don't have a `polya.annot.table` so the notebook errors out. So I suggest we might have to update the 02-analyze-drops.Rmd (to make it more flexible to to able to use only stranded data as we have in kfnbl ) to run per `strategy` like:

```
for strategy in ${libraryStrategies[@]}; do

  Rscript --vanilla 01-summarize_matrices.R \
    -i ../../data/${project_acronym}-gene-expression-rsem-${quantificationType}.${strategy}.rds \
    -g ../../data/gencode.v27.primary_assembly.annotation.gtf.gz \
    -m results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed.${strategy}.rds \
    -t results/k${project_acronym}fnbl-gene-expression-rsem-${quantificationType}-collapsed_table.${strategy}.rds

    # run the notebook for analysis of dropped genes
    Rscript -e "rmarkdown::render(input = '02-analyze-drops.Rmd', output_file = paste0('02-analyze-drops-', '${project_acronym}','-${quantificationType}'),params = list(annot.table = 'results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed_table.${quantificationType}',strategy='${strategy}'), clean = TRUE)"
done

```

Reply:
I could not make a pull request to your forked repo, so I've saved the updated 02-analyze-drops.Rmd code here https://gist.github.com/kgaonkar6/b0f982e59857f258b6f10b0aadd03105 to run per strategy and quantification type.

Reply:
Thanks, @kgaonkar6 . I have updated the script.